packages/react-router/modules/Switch.js

```diff
@@ -42,10 +42,28 @@ class Switch extends React.Component {
     const { route } = this.context.router;
     const { children } = this.props;
     const location = this.props.location || route.location;
+    const { match, child } = getMatch(route, location, children);
 
-    let match, child;
-    React.Children.forEach(children, element => {
-      if (match == null && React.isValidElement(element)) {
+    return match
+      ? React.cloneElement(child, { location, computedMatch: match })
+      : null;
+  }
+}
+
+function getMatch(route, location, children) {
```

Comment:
Can we move this within the component? We don't need it externally and then we can access the props/context directly.

Reply:
I can move it within the component sure, but do know that this was a deliberate decision for the sake of efficiency. There is no need to redefine it over and over again within `render`. Especially considering that it does not need access to `this`, `props` or `state`.

Please let me know whether you still would like this to be inlined.

Reply:
@bripkens moving it within the component doesn't necessarily mean defining it within render, right? It could go on the instance, or possibly as a static class method if React Router transpiles those down.

Reply:
Whoops, I thought the component was just a function when I looked at it again on GitHub. Sorry for that! Changed it accordingly.